plugins/woocommerce/src/StoreApi/Formatters/MoneyFormatter.php

```diff
@@ -23,12 +24,30 @@ public function format( $value, array $options = [] ) {
 			]
 		);
 
-		return (string) intval(
-			round(
-				( (float) wc_format_decimal( $value ) ) * ( 10 ** absint( $options['decimals'] ) ),
-				0,
-				absint( $options['rounding_mode'] )
-			)
-		);
+		// Ensure rounding mode is valid.
+		$rounding_modes           = [ PHP_ROUND_HALF_UP, PHP_ROUND_HALF_DOWN, PHP_ROUND_HALF_EVEN, PHP_ROUND_HALF_ODD ];
+		$options['rounding_mode'] = absint( $options['rounding_mode'] );
+		if ( ! in_array( $options['rounding_mode'], $rounding_modes, true ) ) {
+			$options['rounding_mode'] = PHP_ROUND_HALF_UP;
+		}
+
+		if ( ! is_int( $value ) && ! is_string( $value ) && ! is_float( $value ) ) {
+			wc_doing_it_wrong(
```

Comment:
Another non-essential suggestion (feel free to discard), we could describe this behaviour with an addition test or assertion.

```php
$this->assertEquals( 
	'', 
	$this->mock_formatter->format( array( 9, 99 ) ),
	'When an unexpected type such as an array is passed, the result will be an empty string.'
);
```

(☝🏼 Not tested, just illustrative.)

Reply:
Or perhaps `$this->expectDeprecated()` ...

Reply:
Actually, I did add tests for this, but I'm glad you brought this up because I forgot to rename the test function.